src/components/editor/DistributedTransition.js

```diff
@@ -29,26 +29,67 @@ class DistributedTransition extends Component<Props> {
 
     return (
       <label>
-
         Distributed Transition:
         {currentValue.map((t, i) => {
           return <div key={i} className="transition-option">
             <label>To:
               <RIESelect className='editable-text' propName='transition' value={{id:t.to, text:t.to}} change={this.props.onChange(`[${i}].transition`)} options={options} />
             </label>
             <br/>
-            <label> Weight:
-            <RIENumber className='editable-text' value={t.distribution} propName='distribution' change={this.props.onChange(`[${i}].distribution`)} />
-
-            </label>
+            {this.renderDistribution(t.distribution, i)}
             <br/>
             <a className='editable-text delete-button' onClick={() => this.props.onChange(`[${i}]`)({val: {id: null}})}>remove</a>
           </div>
         })}
         <a className='editable-text add-button' onClick={() => this.props.onChange(`[${currentValue.length}]`)({val: {id: _.cloneDeep(TransitionTemplates.Distributed[0])}})}>+</a>
+        <br/>
+        {this.renderWarning()}
       </label>
     );
   }
+
+  renderDistribution(distribution: mixed, index: number) {
+    if(typeof distribution === 'object') {
+      return (
+        <label>
+          Attribute: <RIEInput className='editable-text' value={distribution.attribute} propName='attribute' change={this.props.onChange(`[${index}].distribution.attribute`)} />
+          <br />
+          Weight: <RIENumber className='editable-text' value={distribution.default} propName='default' editProps={{step: .01, min: 0, max: 1}} format={this.formatAsPercentage} validate={this.checkInRange} change={this.props.onChange(`[${index}].distribution.default`)} />
+          <br />
+          <a className='editable-text' onClick={() => this.props.onChange(`[${index}].distribution`)({val: {id: _.cloneDeep(AttributeTemplates.UnnamedDistribution)}})}>Change to Unnamed Distribution</a>
+        </label>
+      );
+    } else {
+      return (
+        <label> Weight:
+          <RIENumber className='editable-text' value={distribution} propName='distribution' editProps={{step: .01, min: 0, max: 1}} format={this.formatAsPercentage} validate={this.checkInRange} change={this.props.onChange(`[${index}].distribution`)} />
+          <br />
+          <a className='editable-text' onClick={() => this.props.onChange(`[${index}].distribution`)({val: {id: _.cloneDeep(AttributeTemplates.NamedDistribution)}})}>Change to Named Distribution</a>
+        </label>
+      );
+    }
+  }
+
+  renderWarning() {
+    if(!this.props.transition) {
+      return null;
+    }
+    // TODO remove Number calls when it can be ensured the values are numbers
+    let warn = (this.props.transition.transition.reduce((acc, val) => Number(acc) + Number(typeof val.distribution === 'object' ? val.distribution.default : val.distribution), 0) !== 1);
```

Comment:
The `Number` calls should be removed after issue #103 is resolved. Until then, they are necessary for this warning system to function correctly.